safep/AFEP_parse.py

```
@@ -40,13 +40,17 @@ def do_agg_data(dataax, plotax):
     for line in lines:
         agg_data.append(line.get_ydata())
     flat = np.array(agg_data).flatten()
-    kernel = sp.stats.gaussian_kde(flat)
-    pdf_x = np.linspace(-1, 1, 1000)
-    pdf_y = kernel(pdf_x)
+    if np.allclose(flat, 0):
+        warnings.warn("Kernel density estimation failed. All data are 0.")
```

Ruff (0.16.0)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)